libcheckisomd5.c

```diff
@@ -32,336 +31,120 @@
 
 #include "md5.h"
 #include "libcheckisomd5.h"
-
-#define APPDATA_OFFSET 883
-#define SIZE_OFFSET 84
-
-/* Length in characters of string used for fragment md5sum checking */
-#define FRAGMENT_SUM_LENGTH 60
-
-#define MAX(x, y)  ((x > y) ? x : y)
-#define MIN(x, y)  ((x < y) ? x : y)
-
-/* finds primary volume descriptor and returns info from it */
-/* mediasum must be a preallocated buffer at least 33 bytes long */
-/* fragmentsums must be a preallocated buffer at least FRAGMENT_SUM_LENGTH+1 bytes long */
-static int parsepvd(int isofd, char *mediasum, int *skipsectors, long long *isosize, int *supported, char *fragmentsums, long long *fragmentcount) {
-    unsigned char buf[2048];
-    char buf2[512];
-    char tmpbuf[512];
-    int skipfnd, md5fnd, supportedfnd, fragsumfnd, fragcntfnd;
-    unsigned int loc;
-    long long offset;
-    char *p;
-
-    if (lseek(isofd, (off_t)(16L * 2048L), SEEK_SET) == -1)
-        return ((long long)-1);
-
-    offset = (16L * 2048L);
-    for (;1;) {
-        if (read(isofd, buf, 2048) <= 0)
-            return ((long long)-1);
-
-        if (buf[0] == 1)
-            /* found primary volume descriptor */
-            break;
-        else if (buf[0] == 255)
-            /* hit end and didn't find primary volume descriptor */
-            return ((long long)-1);
-        offset += 2048L;
+#include "utilities.h"
+
+static void clear_appdata(unsigned char *const buffer, const size_t size, const off_t appdata_offset, const off_t offset) {
+    static const ssize_t buffer_start = 0;
+    const ssize_t difference = appdata_offset - offset;
+    if (-APPDATA_SIZE <= difference && difference <= (ssize_t) size) {
```

Comment:
`difference` might be negative.

Therefore if I'd flip it I would still have to negate `APPDATA_SIZE`: `difference >= -APPDATA_SIZE`.

I wrote it that way because when I developed this at my whiteboard it said: `-APPDATA_SIZE <= difference <= size` and I thought it's more clear like that in code since it indicates that difference should be in between those two values.